app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeViewModel.kt

```diff
@@ -83,24 +91,29 @@ class HomeViewModel @Inject constructor(
         _showMyCoinsLoading.value = false
     }
 
-    private fun getTrendingCoins() = execute {
-        _showTrendingCoinsLoading.value = true
-        getTrendingCoinsUseCase.execute(
-            GetTrendingCoinsUseCase.Input(
-                currency = FIAT_CURRENCY,
-                order = MY_COINS_ORDER,
-                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
-                itemPerPage = MY_COINS_ITEM_PER_PAGE,
-                page = MY_COINS_INITIAL_PAGE
+    override fun getTrendingCoins(loadMore: Boolean) {
+        if (_showTrendingCoinsLoading.value != LoadingState.Idle) return
+        execute {
+            _showTrendingCoinsLoading.value = if (loadMore) LoadingState.LoadingMore else LoadingState.Loading
+            getTrendingCoinsUseCase.execute(
+                GetTrendingCoinsUseCase.Input(
+                    currency = FIAT_CURRENCY,
+                    order = MY_COINS_ORDER,
+                    priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
+                    itemPerPage = MY_COINS_ITEM_PER_PAGE,
+                    page = page
+                )
             )
-        )
-            .catch { e ->
-                _error.emit(e)
-            }
-            .collect { coins ->
-                _trendingCoins.emit(coins.map { it.toUiModel() })
-            }
-        _showTrendingCoinsLoading.value = false
+                .catch { e ->
+                    _error.emit(e)
+                }
+                .collect { coins ->
+                    val newList = coins.map { it.toUiModel() }
```

Comment:
Should it be `nextPage` or `nextCoinList` 🤔? `newList` sounds like a replacement.